source/common/stats/hystrix.cc

```diff
@@ -0,0 +1,315 @@
+#include "common/stats/hystrix.h"
+
+#include <chrono>
+#include <ctime>
+#include <iostream>
+#include <sstream>
+
+#include "absl/strings/str_cat.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Stats {
+
+const uint64_t Hystrix::DEFAULT_NUM_OF_BUCKETS;
+const uint64_t Hystrix::ROLLING_WINDOW_IN_MS;
+const uint64_t Hystrix::PING_INTERVAL_IN_MS;
+
+// add new value to rolling window, in place of oldest one
+void Hystrix::pushNewValue(std::string key, uint64_t value) {
+  // create vector if do not exist
+  if (rolling_stats_map_.find(key) == rolling_stats_map_.end()) {
+    rolling_stats_map_[key].resize(num_of_buckets_, value);
```

Comment:
rolling_stats_map_[key].resize(num_o_buckets_, value) is all you need; no need for the if-statement or .find() check.